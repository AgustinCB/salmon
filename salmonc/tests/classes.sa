class DevonshireCream {
  serveOn() {
    return "Scones";
  }
  serveWith(condiment) {
    return condiment;
  }
}

print DevonshireCream;
var obj = DevonshireCream();
obj.prop = "test";
print obj;
print obj.prop;
print DevonshireCream.serveOn();
print obj.serveOn();
print DevonshireCream.serveOn;
print obj.serveOn;
print obj.serveWith("cream");
