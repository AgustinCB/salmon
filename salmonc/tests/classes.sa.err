Constants: [SimpleValue(Function { ip: 370, arity: 1, uplifts: None }), SimpleValue(String(0)), SimpleValue(Function { ip: 372, arity: 2, uplifts: None }), SimpleValue(Function { ip: 374, arity: 1, uplifts: None }), SimpleValue(String(6)), SimpleValue(String(14)), SimpleValue(String(20)), SimpleValue(String(36)), SimpleValue(Function { ip: 397, arity: 2, uplifts: None }), SimpleValue(String(52)), SimpleValue(String(68)), SimpleValue(String(78)), SimpleValue(String(85)), SimpleValue(String(95)), SimpleValue(String(104)), SimpleValue(String(114)), SimpleValue(String(122)), SimpleValue(String(138)), SimpleValue(String(148)), SimpleValue(Object { address: 164, tags: 372 }), SimpleValue(String(379)), SimpleValue(Integer(1)), SimpleValue(Integer(4)), SimpleValue(Integer(3)), SimpleValue(String(386)), SimpleValue(String(387)), SimpleValue(String(391)), SimpleValue(String(403)), SimpleValue(String(407)), SimpleValue(String(419)), SimpleValue(String(431)), SimpleValue(String(443)), SimpleValue(String(459)), SimpleValue(String(464)), SimpleValue(String(477)), SimpleValue(String(492)), SimpleValue(String(507)), SimpleValue(String(512)), SimpleValue(String(529)), SimpleValue(String(546))]
Instructions: [Instruction { instruction_type: Constant(0), location: 1 }, Instruction { instruction_type: SetGlobal(2), location: 1 }, Instruction { instruction_type: Pop, location: 1 }, Instruction { instruction_type: Constant(2), location: 2 }, Instruction { instruction_type: SetGlobal(3), location: 2 }, Instruction { instruction_type: Pop, location: 2 }, Instruction { instruction_type: Constant(3), location: 3 }, Instruction { instruction_type: SetGlobal(4), location: 3 }, Instruction { instruction_type: Pop, location: 3 }, Instruction { instruction_type: Constant(8), location: 5 }, Instruction { instruction_type: SetGlobal(5), location: 5 }, Instruction { instruction_type: Pop, location: 5 }, Instruction { instruction_type: Constant(20), location: 5 }, Instruction { instruction_type: Constant(5), location: 5 }, Instruction { instruction_type: Constant(19), location: 5 }, Instruction { instruction_type: AddTag, location: 5 }, Instruction { instruction_type: AddTag, location: 5 }, Instruction { instruction_type: SetGlobal(6), location: 5 }, Instruction { instruction_type: Pop, location: 5 }, Instruction { instruction_type: Constant(24), location: 6 }, Instruction { instruction_type: GetGlobal(6), location: 6 }, Instruction { instruction_type: ToStr, location: 6 }, Instruction { instruction_type: StringConcat, location: 6 }, Instruction { instruction_type: Push, location: 6 }, Instruction { instruction_type: Strlen, location: 6 }, Instruction { instruction_type: Swap, location: 6 }, Instruction { instruction_type: Constant(21), location: 6 }, Instruction { instruction_type: Constant(23), location: 6 }, Instruction { instruction_type: Constant(22), location: 6 }, Instruction { instruction_type: Syscall, location: 6 }, Instruction { instruction_type: Pop, location: 6 }, Instruction { instruction_type: GetGlobal(6), location: 7 }, Instruction { instruction_type: Duplicate, location: 7 }, Instruction { instruction_type: CheckType(7), location: 7 }, Instruction { instruction_type: JmpIfFalse(3), location: 7 }, Instruction { instruction_type: Constant(5), location: 7 }, Instruction { instruction_type: Swap, location: 7 }, Instruction { instruction_type: RemoveTag, location: 7 }, Instruction { instruction_type: Call, location: 7 }, Instruction { instruction_type: SetGlobal(0), location: 7 }, Instruction { instruction_type: Pop, location: 7 }, Instruction { instruction_type: Constant(27), location: 8 }, Instruction { instruction_type: Constant(26), location: 8 }, Instruction { instruction_type: GetGlobal(0), location: 8 }, Instruction { instruction_type: ObjectHas, location: 8 }, Instruction { instruction_type: JmpIfFalse(5), location: 8 }, Instruction { instruction_type: Constant(26), location: 8 }, Instruction { instruction_type: Swap, location: 8 }, Instruction { instruction_type: ObjectGet, location: 8 }, Instruction { instruction_type: Call, location: 8 }, Instruction { instruction_type: Jmp(4), location: 8 }, Instruction { instruction_type: Constant(25), location: 8 }, Instruction { instruction_type: Swap, location: 8 }, Instruction { instruction_type: ObjectSet, location: 8 }, Instruction { instruction_type: Pop, location: 8 }, Instruction { instruction_type: Pop, location: 8 }, Instruction { instruction_type: Constant(28), location: 9 }, Instruction { instruction_type: Constant(18), location: 9 }, Instruction { instruction_type: GetGlobal(0), location: 9 }, Instruction { instruction_type: ObjectHas, location: 9 }, Instruction { instruction_type: JmpIfFalse(5), location: 9 }, Instruction { instruction_type: Constant(18), location: 9 }, Instruction { instruction_type: Swap, location: 9 }, Instruction { instruction_type: ObjectGet, location: 9 }, Instruction { instruction_type: Call, location: 9 }, Instruction { instruction_type: Jmp(4), location: 9 }, Instruction { instruction_type: Constant(15), location: 9 }, Instruction { instruction_type: Swap, location: 9 }, Instruction { instruction_type: ObjectSet, location: 9 }, Instruction { instruction_type: Pop, location: 9 }, Instruction { instruction_type: Pop, location: 9 }, Instruction { instruction_type: GetGlobal(6), location: 10 }, Instruction { instruction_type: Duplicate, location: 10 }, Instruction { instruction_type: CheckType(7), location: 10 }, Instruction { instruction_type: JmpIfFalse(3), location: 10 }, Instruction { instruction_type: Constant(5), location: 10 }, Instruction { instruction_type: Swap, location: 10 }, Instruction { instruction_type: RemoveTag, location: 10 }, Instruction { instruction_type: Call, location: 10 }, Instruction { instruction_type: SetGlobal(1), location: 10 }, Instruction { instruction_type: Pop, location: 10 }, Instruction { instruction_type: Constant(24), location: 11 }, Instruction { instruction_type: GetGlobal(0), location: 11 }, Instruction { instruction_type: ToStr, location: 11 }, Instruction { instruction_type: StringConcat, location: 11 }, Instruction { instruction_type: Push, location: 11 }, Instruction { instruction_type: Strlen, location: 11 }, Instruction { instruction_type: Swap, location: 11 }, Instruction { instruction_type: Constant(21), location: 11 }, Instruction { instruction_type: Constant(23), location: 11 }, Instruction { instruction_type: Constant(22), location: 11 }, Instruction { instruction_type: Syscall, location: 11 }, Instruction { instruction_type: Pop, location: 11 }, Instruction { instruction_type: Constant(24), location: 12 }, Instruction { instruction_type: GetGlobal(0), location: 12 }, Instruction { instruction_type: Duplicate, location: 12 }, Instruction { instruction_type: Constant(5), location: 12 }, Instruction { instruction_type: Swap, location: 12 }, Instruction { instruction_type: CheckTag, location: 12 }, Instruction { instruction_type: JmpIfFalse(4), location: 12 }, Instruction { instruction_type: Constant(29), location: 12 }, Instruction { instruction_type: Swap, location: 12 }, Instruction { instruction_type: ObjectGet, location: 12 }, Instruction { instruction_type: Jmp(12), location: 12 }, Instruction { instruction_type: Constant(30), location: 12 }, Instruction { instruction_type: Swap, location: 12 }, Instruction { instruction_type: ObjectHas, location: 12 }, Instruction { instruction_type: JmpIfFalse(5), location: 12 }, Instruction { instruction_type: Constant(30), location: 12 }, Instruction { instruction_type: Swap, location: 12 }, Instruction { instruction_type: ObjectGet, location: 12 }, Instruction { instruction_type: Call, location: 12 }, Instruction { instruction_type: Jmp(3), location: 12 }, Instruction { instruction_type: Constant(25), location: 12 }, Instruction { instruction_type: Swap, location: 12 }, Instruction { instruction_type: ObjectGet, location: 12 }, Instruction { instruction_type: ToStr, location: 12 }, Instruction { instruction_type: StringConcat, location: 12 }, Instruction { instruction_type: Push, location: 12 }, Instruction { instruction_type: Strlen, location: 12 }, Instruction { instruction_type: Swap, location: 12 }, Instruction { instruction_type: Constant(21), location: 12 }, Instruction { instruction_type: Constant(23), location: 12 }, Instruction { instruction_type: Constant(22), location: 12 }, Instruction { instruction_type: Syscall, location: 12 }, Instruction { instruction_type: Pop, location: 12 }, Instruction { instruction_type: Constant(24), location: 13 }, Instruction { instruction_type: GetGlobal(0), location: 13 }, Instruction { instruction_type: Duplicate, location: 13 }, Instruction { instruction_type: Constant(5), location: 13 }, Instruction { instruction_type: Swap, location: 13 }, Instruction { instruction_type: CheckTag, location: 13 }, Instruction { instruction_type: JmpIfFalse(4), location: 13 }, Instruction { instruction_type: Constant(31), location: 13 }, Instruction { instruction_type: Swap, location: 13 }, Instruction { instruction_type: ObjectGet, location: 13 }, Instruction { instruction_type: Jmp(12), location: 13 }, Instruction { instruction_type: Constant(16), location: 13 }, Instruction { instruction_type: Swap, location: 13 }, Instruction { instruction_type: ObjectHas, location: 13 }, Instruction { instruction_type: JmpIfFalse(5), location: 13 }, Instruction { instruction_type: Constant(16), location: 13 }, Instruction { instruction_type: Swap, location: 13 }, Instruction { instruction_type: ObjectGet, location: 13 }, Instruction { instruction_type: Call, location: 13 }, Instruction { instruction_type: Jmp(3), location: 13 }, Instruction { instruction_type: Constant(15), location: 13 }, Instruction { instruction_type: Swap, location: 13 }, Instruction { instruction_type: ObjectGet, location: 13 }, Instruction { instruction_type: ToStr, location: 13 }, Instruction { instruction_type: StringConcat, location: 13 }, Instruction { instruction_type: Push, location: 13 }, Instruction { instruction_type: Strlen, location: 13 }, Instruction { instruction_type: Swap, location: 13 }, Instruction { instruction_type: Constant(21), location: 13 }, Instruction { instruction_type: Constant(23), location: 13 }, Instruction { instruction_type: Constant(22), location: 13 }, Instruction { instruction_type: Syscall, location: 13 }, Instruction { instruction_type: Pop, location: 13 }, Instruction { instruction_type: Constant(32), location: 14 }, Instruction { instruction_type: Constant(26), location: 14 }, Instruction { instruction_type: GetGlobal(1), location: 14 }, Instruction { instruction_type: ObjectHas, location: 14 }, Instruction { instruction_type: JmpIfFalse(5), location: 14 }, Instruction { instruction_type: Constant(26), location: 14 }, Instruction { instruction_type: Swap, location: 14 }, Instruction { instruction_type: ObjectGet, location: 14 }, Instruction { instruction_type: Call, location: 14 }, Instruction { instruction_type: Jmp(4), location: 14 }, Instruction { instruction_type: Constant(25), location: 14 }, Instruction { instruction_type: Swap, location: 14 }, Instruction { instruction_type: ObjectSet, location: 14 }, Instruction { instruction_type: Pop, location: 14 }, Instruction { instruction_type: Pop, location: 14 }, Instruction { instruction_type: Constant(33), location: 15 }, Instruction { instruction_type: Constant(18), location: 15 }, Instruction { instruction_type: GetGlobal(1), location: 15 }, Instruction { instruction_type: ObjectHas, location: 15 }, Instruction { instruction_type: JmpIfFalse(5), location: 15 }, Instruction { instruction_type: Constant(18), location: 15 }, Instruction { instruction_type: Swap, location: 15 }, Instruction { instruction_type: ObjectGet, location: 15 }, Instruction { instruction_type: Call, location: 15 }, Instruction { instruction_type: Jmp(4), location: 15 }, Instruction { instruction_type: Constant(15), location: 15 }, Instruction { instruction_type: Swap, location: 15 }, Instruction { instruction_type: ObjectSet, location: 15 }, Instruction { instruction_type: Pop, location: 15 }, Instruction { instruction_type: Pop, location: 15 }, Instruction { instruction_type: Constant(24), location: 16 }, Instruction { instruction_type: GetGlobal(1), location: 16 }, Instruction { instruction_type: Duplicate, location: 16 }, Instruction { instruction_type: Constant(5), location: 16 }, Instruction { instruction_type: Swap, location: 16 }, Instruction { instruction_type: CheckTag, location: 16 }, Instruction { instruction_type: JmpIfFalse(4), location: 16 }, Instruction { instruction_type: Constant(29), location: 16 }, Instruction { instruction_type: Swap, location: 16 }, Instruction { instruction_type: ObjectGet, location: 16 }, Instruction { instruction_type: Jmp(12), location: 16 }, Instruction { instruction_type: Constant(30), location: 16 }, Instruction { instruction_type: Swap, location: 16 }, Instruction { instruction_type: ObjectHas, location: 16 }, Instruction { instruction_type: JmpIfFalse(5), location: 16 }, Instruction { instruction_type: Constant(30), location: 16 }, Instruction { instruction_type: Swap, location: 16 }, Instruction { instruction_type: ObjectGet, location: 16 }, Instruction { instruction_type: Call, location: 16 }, Instruction { instruction_type: Jmp(3), location: 16 }, Instruction { instruction_type: Constant(25), location: 16 }, Instruction { instruction_type: Swap, location: 16 }, Instruction { instruction_type: ObjectGet, location: 16 }, Instruction { instruction_type: ToStr, location: 16 }, Instruction { instruction_type: StringConcat, location: 16 }, Instruction { instruction_type: Push, location: 16 }, Instruction { instruction_type: Strlen, location: 16 }, Instruction { instruction_type: Swap, location: 16 }, Instruction { instruction_type: Constant(21), location: 16 }, Instruction { instruction_type: Constant(23), location: 16 }, Instruction { instruction_type: Constant(22), location: 16 }, Instruction { instruction_type: Syscall, location: 16 }, Instruction { instruction_type: Pop, location: 16 }, Instruction { instruction_type: Constant(24), location: 17 }, Instruction { instruction_type: GetGlobal(1), location: 17 }, Instruction { instruction_type: Duplicate, location: 17 }, Instruction { instruction_type: Constant(5), location: 17 }, Instruction { instruction_type: Swap, location: 17 }, Instruction { instruction_type: CheckTag, location: 17 }, Instruction { instruction_type: JmpIfFalse(4), location: 17 }, Instruction { instruction_type: Constant(31), location: 17 }, Instruction { instruction_type: Swap, location: 17 }, Instruction { instruction_type: ObjectGet, location: 17 }, Instruction { instruction_type: Jmp(12), location: 17 }, Instruction { instruction_type: Constant(16), location: 17 }, Instruction { instruction_type: Swap, location: 17 }, Instruction { instruction_type: ObjectHas, location: 17 }, Instruction { instruction_type: JmpIfFalse(5), location: 17 }, Instruction { instruction_type: Constant(16), location: 17 }, Instruction { instruction_type: Swap, location: 17 }, Instruction { instruction_type: ObjectGet, location: 17 }, Instruction { instruction_type: Call, location: 17 }, Instruction { instruction_type: Jmp(3), location: 17 }, Instruction { instruction_type: Constant(15), location: 17 }, Instruction { instruction_type: Swap, location: 17 }, Instruction { instruction_type: ObjectGet, location: 17 }, Instruction { instruction_type: ToStr, location: 17 }, Instruction { instruction_type: StringConcat, location: 17 }, Instruction { instruction_type: Push, location: 17 }, Instruction { instruction_type: Strlen, location: 17 }, Instruction { instruction_type: Swap, location: 17 }, Instruction { instruction_type: Constant(21), location: 17 }, Instruction { instruction_type: Constant(23), location: 17 }, Instruction { instruction_type: Constant(22), location: 17 }, Instruction { instruction_type: Syscall, location: 17 }, Instruction { instruction_type: Pop, location: 17 }, Instruction { instruction_type: Constant(24), location: 18 }, Instruction { instruction_type: GetGlobal(0), location: 18 }, Instruction { instruction_type: Duplicate, location: 18 }, Instruction { instruction_type: Constant(5), location: 18 }, Instruction { instruction_type: Swap, location: 18 }, Instruction { instruction_type: CheckTag, location: 18 }, Instruction { instruction_type: JmpIfFalse(4), location: 18 }, Instruction { instruction_type: Constant(34), location: 18 }, Instruction { instruction_type: Swap, location: 18 }, Instruction { instruction_type: ObjectGet, location: 18 }, Instruction { instruction_type: Jmp(12), location: 18 }, Instruction { instruction_type: Constant(35), location: 18 }, Instruction { instruction_type: Swap, location: 18 }, Instruction { instruction_type: ObjectHas, location: 18 }, Instruction { instruction_type: JmpIfFalse(5), location: 18 }, Instruction { instruction_type: Constant(35), location: 18 }, Instruction { instruction_type: Swap, location: 18 }, Instruction { instruction_type: ObjectGet, location: 18 }, Instruction { instruction_type: Call, location: 18 }, Instruction { instruction_type: Jmp(3), location: 18 }, Instruction { instruction_type: Constant(11), location: 18 }, Instruction { instruction_type: Swap, location: 18 }, Instruction { instruction_type: ObjectGet, location: 18 }, Instruction { instruction_type: Duplicate, location: 18 }, Instruction { instruction_type: CheckType(7), location: 18 }, Instruction { instruction_type: JmpIfFalse(3), location: 18 }, Instruction { instruction_type: Constant(5), location: 18 }, Instruction { instruction_type: Swap, location: 18 }, Instruction { instruction_type: RemoveTag, location: 18 }, Instruction { instruction_type: Call, location: 18 }, Instruction { instruction_type: ToStr, location: 18 }, Instruction { instruction_type: StringConcat, location: 18 }, Instruction { instruction_type: Push, location: 18 }, Instruction { instruction_type: Strlen, location: 18 }, Instruction { instruction_type: Swap, location: 18 }, Instruction { instruction_type: Constant(21), location: 18 }, Instruction { instruction_type: Constant(23), location: 18 }, Instruction { instruction_type: Constant(22), location: 18 }, Instruction { instruction_type: Syscall, location: 18 }, Instruction { instruction_type: Pop, location: 18 }, Instruction { instruction_type: Constant(24), location: 19 }, Instruction { instruction_type: GetGlobal(0), location: 19 }, Instruction { instruction_type: Duplicate, location: 19 }, Instruction { instruction_type: Constant(5), location: 19 }, Instruction { instruction_type: Swap, location: 19 }, Instruction { instruction_type: CheckTag, location: 19 }, Instruction { instruction_type: JmpIfFalse(4), location: 19 }, Instruction { instruction_type: Constant(34), location: 19 }, Instruction { instruction_type: Swap, location: 19 }, Instruction { instruction_type: ObjectGet, location: 19 }, Instruction { instruction_type: Jmp(12), location: 19 }, Instruction { instruction_type: Constant(35), location: 19 }, Instruction { instruction_type: Swap, location: 19 }, Instruction { instruction_type: ObjectHas, location: 19 }, Instruction { instruction_type: JmpIfFalse(5), location: 19 }, Instruction { instruction_type: Constant(35), location: 19 }, Instruction { instruction_type: Swap, location: 19 }, Instruction { instruction_type: ObjectGet, location: 19 }, Instruction { instruction_type: Call, location: 19 }, Instruction { instruction_type: Jmp(3), location: 19 }, Instruction { instruction_type: Constant(11), location: 19 }, Instruction { instruction_type: Swap, location: 19 }, Instruction { instruction_type: ObjectGet, location: 19 }, Instruction { instruction_type: ToStr, location: 19 }, Instruction { instruction_type: StringConcat, location: 19 }, Instruction { instruction_type: Push, location: 19 }, Instruction { instruction_type: Strlen, location: 19 }, Instruction { instruction_type: Swap, location: 19 }, Instruction { instruction_type: Constant(21), location: 19 }, Instruction { instruction_type: Constant(23), location: 19 }, Instruction { instruction_type: Constant(22), location: 19 }, Instruction { instruction_type: Syscall, location: 19 }, Instruction { instruction_type: Pop, location: 19 }, Instruction { instruction_type: Constant(24), location: 20 }, Instruction { instruction_type: Constant(36), location: 20 }, Instruction { instruction_type: GetGlobal(0), location: 20 }, Instruction { instruction_type: Duplicate, location: 20 }, Instruction { instruction_type: Constant(5), location: 20 }, Instruction { instruction_type: Swap, location: 20 }, Instruction { instruction_type: CheckTag, location: 20 }, Instruction { instruction_type: JmpIfFalse(4), location: 20 }, Instruction { instruction_type: Constant(37), location: 20 }, Instruction { instruction_type: Swap, location: 20 }, Instruction { instruction_type: ObjectGet, location: 20 }, Instruction { instruction_type: Jmp(12), location: 20 }, Instruction { instruction_type: Constant(38), location: 20 }, Instruction { instruction_type: Swap, location: 20 }, Instruction { instruction_type: ObjectHas, location: 20 }, Instruction { instruction_type: JmpIfFalse(5), location: 20 }, Instruction { instruction_type: Constant(38), location: 20 }, Instruction { instruction_type: Swap, location: 20 }, Instruction { instruction_type: ObjectGet, location: 20 }, Instruction { instruction_type: Call, location: 20 }, Instruction { instruction_type: Jmp(3), location: 20 }, Instruction { instruction_type: Constant(13), location: 20 }, Instruction { instruction_type: Swap, location: 20 }, Instruction { instruction_type: ObjectGet, location: 20 }, Instruction { instruction_type: Duplicate, location: 20 }, Instruction { instruction_type: CheckType(7), location: 20 }, Instruction { instruction_type: JmpIfFalse(3), location: 20 }, Instruction { instruction_type: Constant(5), location: 20 }, Instruction { instruction_type: Swap, location: 20 }, Instruction { instruction_type: RemoveTag, location: 20 }, Instruction { instruction_type: Call, location: 20 }, Instruction { instruction_type: ToStr, location: 20 }, Instruction { instruction_type: StringConcat, location: 20 }, Instruction { instruction_type: Push, location: 20 }, Instruction { instruction_type: Strlen, location: 20 }, Instruction { instruction_type: Swap, location: 20 }, Instruction { instruction_type: Constant(21), location: 20 }, Instruction { instruction_type: Constant(23), location: 20 }, Instruction { instruction_type: Constant(22), location: 20 }, Instruction { instruction_type: Syscall, location: 20 }, Instruction { instruction_type: Pop, location: 20 }, Instruction { instruction_type: Return, location: 20 }, Instruction { instruction_type: Constant(1), location: 1 }, Instruction { instruction_type: Return, location: 1 }, Instruction { instruction_type: GetLocal(1), location: 2 }, Instruction { instruction_type: Return, location: 2 }, Instruction { instruction_type: GetLocal(0), location: 3 }, Instruction { instruction_type: Duplicate, location: 3 }, Instruction { instruction_type: Constant(5), location: 3 }, Instruction { instruction_type: Swap, location: 3 }, Instruction { instruction_type: CheckTag, location: 3 }, Instruction { instruction_type: JmpIfFalse(4), location: 3 }, Instruction { instruction_type: Constant(6), location: 3 }, Instruction { instruction_type: Swap, location: 3 }, Instruction { instruction_type: ObjectGet, location: 3 }, Instruction { instruction_type: Jmp(12), location: 3 }, Instruction { instruction_type: Constant(7), location: 3 }, Instruction { instruction_type: Swap, location: 3 }, Instruction { instruction_type: ObjectHas, location: 3 }, Instruction { instruction_type: JmpIfFalse(5), location: 3 }, Instruction { instruction_type: Constant(7), location: 3 }, Instruction { instruction_type: Swap, location: 3 }, Instruction { instruction_type: ObjectGet, location: 3 }, Instruction { instruction_type: Call, location: 3 }, Instruction { instruction_type: Jmp(3), location: 3 }, Instruction { instruction_type: Constant(4), location: 3 }, Instruction { instruction_type: Swap, location: 3 }, Instruction { instruction_type: ObjectGet, location: 3 }, Instruction { instruction_type: Return, location: 3 }, Instruction { instruction_type: GetLocal(1), location: 4 }, Instruction { instruction_type: Constant(9), location: 4 }, Instruction { instruction_type: GetLocal(0), location: 4 }, Instruction { instruction_type: ObjectHas, location: 4 }, Instruction { instruction_type: JmpIfFalse(5), location: 4 }, Instruction { instruction_type: Constant(9), location: 4 }, Instruction { instruction_type: Swap, location: 4 }, Instruction { instruction_type: ObjectGet, location: 4 }, Instruction { instruction_type: Call, location: 4 }, Instruction { instruction_type: Jmp(4), location: 4 }, Instruction { instruction_type: Constant(4), location: 4 }, Instruction { instruction_type: Swap, location: 4 }, Instruction { instruction_type: ObjectSet, location: 4 }, Instruction { instruction_type: Pop, location: 4 }, Instruction { instruction_type: Pop, location: 4 }, Instruction { instruction_type: GetLocal(1), location: 5 }, Instruction { instruction_type: Return, location: 5 }]
Locations: [Location { address: 546, line: 0 }, Location { address: 546, line: 2 }, Location { address: 546, line: 5 }, Location { address: 546, line: 8 }, Location { address: 546, line: 11 }, Location { address: 546, line: 12 }, Location { address: 546, line: 16 }, Location { address: 546, line: 17 }, Location { address: 546, line: 18 }, Location { address: 546, line: 19 }, Location { address: 546, line: 20 }, Location { address: 546, line: 21 }, Location { address: 546, line: 22 }, Location { address: 546, line: 23 }, Location { address: 546, line: 24 }, Location { address: 546, line: 25 }, Location { address: 546, line: 26 }, Location { address: 546, line: 27 }, Location { address: 546, line: 28 }, Location { address: 546, line: 29 }, Location { address: 546, line: 30 }]
