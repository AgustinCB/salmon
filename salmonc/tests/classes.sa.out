[object 164]
[object 812]
test
testProp 111
test1
testProp 1111
Scones
[partial function]
cream
